torch/csrc/jit/python_arg_flatten.cpp

```diff
@@ -42,14 +46,37 @@ void flatten_rec(PyObject* obj, ParsedArgs& args) {
     for (auto item : py::reinterpret_borrow<py::list>(obj))
       flatten_rec(item.ptr(), args);
     structure.push_back(D::ListClose);
+  } else if (PyDict_Check(obj)) {
+    auto dict_items = PyDict_Items(obj);
+    structure.push_back(D::DictOpen);
+    for (auto item : py::reinterpret_borrow<py::list>(dict_items)){
+      flatten_rec(item.ptr(), args);
+    }
+    structure.push_back(D::DictClose);
+    std::string msg =
```

Comment:
This warning prints each time flatten() is called in the code.
We can keep it this way (and have the warning print multiple times in the code), or I an add a boolean arg to flatten() to only print the warnings if the caller is LegacyTracedModule() (and thus print the warning once).

Reply:
Is the only limitation that the keys must remain the same during the function?

I don't think we should warn at all unless we can confirm that the user is doing something wrong. We don't want to teach people that they should ignore the warnings they see because they are expected. Can we validate after tracing that the dictionary keys are identical?

Reply:
Yes, that's correct, I  will then remove the warnings.

I am not sure to understand your comment "Can we validate after tracing that the dictionary keys are identical?". 
you mean by that, that we should verify that the dictionary keys are unchanged during the export?